src/encoded/types/cart.py

```diff
@@ -85,8 +75,7 @@ def _create_cart(request, user, name=None, identifier=None, status=None):
     return cart_path
 
 
-@view_config(context=Cart.Collection, request_method='GET', permission='save-carts', name='get-cart')
-def get_or_create_cart_by_user(context, request):
+def _get_user_carts(request, blocked_statuses=[]):
```

Comment:
The _get_user_carts versus get_cart_objects_by_user function is still confusing to me. Probably there needs to be a _get_userid function with the first part of this logic: 

```python
userid = [
        p.replace('userid.', '')
        for p in request.effective_principals
        if p.startswith('userid.')
    ]
    if not userid:
        raise HTTPBadRequest()
    else:
        userid = userid[0]
    user = request.registry[COLLECTIONS]['user'].get(userid)
    if not user:
        raise HTTPBadRequest()
```

And the get_cart_objects_by_user can be called in both files (and _get_user_carts can be deleted).

Reply:
I think you’re looking at old code. There is no _get_user_carts, and there is a _get_userid

Reply:
Well that helps with the confusion. Sorry about that.